src/App.tsx

```diff
@@ -1,73 +1,89 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isInputFocused, setIsInputFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
```

Comment:
The `applyQuery` function uses `debounce`, which delays the update of `appliedQuery`. This means that `appliedQuery` might not reflect the latest `query` value immediately. Consider whether this behavior is acceptable for your application's requirements.